core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -436,83 +415,73 @@ public boolean connect(Socket socket) throws IOException {
         }
     }
 
-    /**
-     * Reschedules the {@code TcpSlaveAgentListener} on demand.  Disables itself after running.
-     */
     @Extension
     @Restricted(NoExternalUse.class)
-    public static class TcpSlaveAgentListenerRescheduler extends AperiodicWork {
-        private Thread originThread;
-        private Throwable cause;
-        private long recurrencePeriod = 5000;
-        private boolean isActive;
+    public static final class TcpSlaveAgentListenerRescheduler extends AperiodicWork {
+        private static final long RECURRENCE_PERIOD = 5000L;
+        private static boolean active = false;
 
         public TcpSlaveAgentListenerRescheduler() {
-            isActive = false;
+            active = false;
         }
 
-        public TcpSlaveAgentListenerRescheduler(Thread originThread, Throwable cause) {
-            this.originThread = originThread;
-            this.cause = cause;
-            this.isActive = false;
+        @Deprecated
```

Comment:
I am deprecating those methods because there are `public` although I did not find any usage.. Or is it OK to remove them ?

Reply:
Even if there are no traces in public plugins, there may be some in closed sourced plugins, we cannot track. We can stick to deprecations and possibly add some javadocs outlining why this method is deprecated, if there are people depending on it.

Reply:
I updated the javadocs

Reply:
Allow me to express a different perspective. Yes what Alex wrote is not wrong but it is more complicated than that. We have to distinguish between a few cases:

1. Used by open-source Jenkins plugins: anyone can check the sources and verify whether something is used or not.
2. Used by CloudBees plugins whose sources are not available to the public but whose binaries are: anyone can check whether something is used or not via the binaries as described [here](https://github.com/jenkinsci/jep/blob/master/jep/227/README.adoc#searching-for-api-usages-in-binaries), but this is laborious. CloudBees employees (who are active in this community) can more easily check whether something is used or not in the sources.
3. Used by non-CloudBees closed-source plugins where neither sources nor binaries are available to the public. We have little to no visibility into these.

There is a real cost to maintaining deprecated functionality: just the fact that it exists means that it's easy for someone to accidentally call it and ignore the warning. Not to mention that it makes it much harder to read and understand the code.

Checking case (1) is feasible for anybody. Checking case (2) is easy for a CloudBees employee but feasible (though difficult) for a non-Bee. Checking case (3) is prohibitively difficult.

My opinion is that if (1) and (2) can be verified, and if (3) is unlikely, that we should go ahead and remove the code. The benefit (lower cognitive load) is worth the minuscule risk of (3). In this case, based on the relatively obscure nature of this code, I cannot imagine that (3) would actually be the case. So as long as (1) and (2) are the case, I would recommend deleting this code.